pkg/provisioner/ironic/ironic.go

```diff
@@ -1265,6 +1265,25 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if p.host.Status.PoweredOn {
+		p.log.Info("host is being powered off before removal")
+		provResult, err := p.hardPowerOff()
+		switch {
+		case err != nil:
```

Comment:
Do we care about 409 errors telling us Ironic is busy doing something with the host?

Reply:
I wonder if Ironic will let us delete a host if it thinks it's busy? That error would happen (and could be handled) on the delete call, so I guess the worst thing that might happen is we do successfully delete a host without powering it off.

Reply:
hardPowerOff() internally uses changePower() which already takes care of 409 errors.